iotdb-core/datanode/src/main/java/org/apache/iotdb/db/pipe/task/subtask/connector/PipeRealtimePriorityBlockingQueue.java

```diff
@@ -89,19 +107,26 @@ public Event directPoll() {
   public Event waitedPoll() {
     Event event = null;
 
-    if (!super.isEmpty()) {
+    if (eventCount.get() >= pendingQueueConsumeThreshold) {
+      event = tsfileInsertEventDeque.pollLast();
+      resetEventCount();
+    }
+    if (event == null && !super.isEmpty()) {
       // Sequentially poll the first offered non-TsFileInsertionEvent
       event = super.directPoll();
+      eventCount.incrementAndGet();
     } else if (!tsfileInsertEventDeque.isEmpty()) {
```

Comment:
If the event is not null, the previous event will be discarded and will again be "Deque.pollLast()".....